spi-imx driver needs two clock: ipg and per. The first clock must be named and
the second must be added.

Signed-off-by: Gwenhael Goavec-Merou <gwenhael.goavec-merou@armadeus.com>
---

Index: linux-3.7/arch/arm/mach-imx/clk-imx27.c
===================================================================
--- linux-3.7.orig/arch/arm/mach-imx/clk-imx27.c
+++ linux-3.7/arch/arm/mach-imx/clk-imx27.c
@@ -229,9 +229,12 @@ int __init mx27_clocks_init(unsigned lon
 	clk_register_clkdev(clk[sdhc2_ipg_gate], "ipg", "mxc-mmc.1");
 	clk_register_clkdev(clk[per2_gate], "per", "mxc-mmc.2");
 	clk_register_clkdev(clk[sdhc2_ipg_gate], "ipg", "mxc-mmc.2");
-	clk_register_clkdev(clk[cspi1_ipg_gate], NULL, "imx27-cspi.0");
-	clk_register_clkdev(clk[cspi2_ipg_gate], NULL, "imx27-cspi.1");
-	clk_register_clkdev(clk[cspi3_ipg_gate], NULL, "imx27-cspi.2");
+	clk_register_clkdev(clk[per2_gate], "per", "imx27-cspi.0");
+	clk_register_clkdev(clk[cspi1_ipg_gate], "ipg", "imx27-cspi.0");
+	clk_register_clkdev(clk[per2_gate], "per", "imx27-cspi.1");
+	clk_register_clkdev(clk[cspi2_ipg_gate], "ipg", "imx27-cspi.1");
+	clk_register_clkdev(clk[per2_gate], "per", "imx27-cspi.2");
+	clk_register_clkdev(clk[cspi3_ipg_gate], "ipg", "imx27-cspi.2");
 	clk_register_clkdev(clk[per3_gate], "per", "imx-fb.0");
 	clk_register_clkdev(clk[lcdc_ipg_gate], "ipg", "imx-fb.0");
 	clk_register_clkdev(clk[lcdc_ahb_gate], "ahb", "imx-fb.0");
